packages/sdk/src/chains/EVM/utils/depositFns.ts

```diff
@@ -0,0 +1,296 @@
+ * const depositEvent = await getDepositEventFromReceipt(contractReceipt, bridge);
+ * console.log('Deposit event:', depositEvent);
+ * console.log('Deposit nonce:', depositEvent.args.depositNonce)
+ *
+ * @param {ethers.ContractReceipt} depositTx - The contract receipt containing the deposit transaction details.
+ * @param {Bridge} bridgeContract - The bridge contract instance used to query the deposit event.
+ * @returns {Promise<DepositEvent>} A promise that resolves to the deposit event associated with the given contract receipt.
+ */
+
+export const getDepositEventFromReceipt = async (
+  depositTx: ethers.ContractReceipt,
+  bridgeContract: Bridge,
+): Promise<DepositEvent> => {
+  try {
+    const depositFilter = bridgeContract.filters.Deposit();
+    const events = await bridgeContract.queryFilter(depositFilter, depositTx.blockHash);
+    const event = events[0];
+    return event;
+  } catch (error) {
+    console.error('Error on getDepositEventFromReceipt', error);
+    return Promise.reject(error);
+  }
+};
+
+/**
+ * Processes a token transfer to the bridge on top level of abstracttion, handling both ERC20 and ERC721 tokens.
+ *
+ * @example
+ * // this short example could miss some params, please look at types for correct info
+ * const depositParams = { resourceId: '0x123', amountOrId: "100", recepientAddress: "0x0123", feeData: "// fee data ///" };
+ * const bridgeConfig = { tokens: [{ resourceId: '0x123', address: '0x456', type: 'erc20' }], bridgeAddress: '0x789', domainId: 1 };
+ * const provider = new ethers.providers.Web3Provider(window.ethereum);
+ * // any ovveride settting for etherjs tranasaction
+ * const overrides = { gasLimit: 100000 };
+ * const receipt = await processTokenTranfer({ depositParams, bridgeConfig, provider, overrides });
+ * // use the getDepositEventFromReceipt method to get the depositNonce
+ *
+ * @param {ProcessTokenTranferParamsType} params - The parameters for processing the token transfer.
+ * @returns {Promise<ethers.ContractReceipt>} - A promise that resolves to the transaction receipt once the transfer is complete.
+ */
+export const processTokenTranfer = async ({
+  depositParams,
+  bridgeConfig,
+  provider,
+  overrides,
+}: ProcessTokenTranferParamsType): Promise<ethers.ContractReceipt> => {
+  const { tokens, bridgeAddress, domainId, confirmations: defaultConfirmations } = bridgeConfig;
+  const { resourceId } = depositParams;
+
+  const selectedToken = tokens.find(token => token.resourceId === resourceId);
+
+  if (!selectedToken) {
+    throw Error(`Can't find in networkConfig token with resourceID: ${resourceId}`);
+  }
+
+  const bridgeInstance = Bridge__factory.connect(bridgeAddress, provider);
+  const confirmations = defaultConfirmations ?? 10;
+
+  const commonTransferParams = {
+    ...depositParams,
+    domainId,
+    confirmations,
+    bridgeInstance,
+    provider,
+    overrides,
+  };
+
+  if (selectedToken.type === 'erc721') {
```

Comment:
Minor nitpick, but a switch() would be nicer with a default case that throws an error.